modules/cli/src/main/scala/scala/cli/commands/Package.scala

```diff
@@ -923,10 +924,11 @@ object Package extends ScalaCommand[PackageOptions] {
         nativeWorkDir
       )
 
-    if (cacheData.changed)
+    if (cacheData.changed) {
+      NativeResourceMapper.copyCFilesToScalaNativeDir(build, nativeWorkDir)
       Library.withLibraryJar(build, dest.last.stripSuffix(".jar")) { mainJar =>
 
-        val classpath = build.fullClassPath.map(_.toString) :+ mainJar.toString
+        val classpath = build.artifacts.classPath.map(_.toString) :+ mainJar.toString
```

Comment:
Don't know how Scala Native handles it down the line, but I'll put `mainJar` first just in case, so that its classes are used in priority over those of the dependencies:
```suggestion
        val classpath = mainJar.toString +: build.artifacts.classPath.map(_.toString)
```